### PR TITLE
948 always capitalize first letter

### LINKED ISSUE #948: MoveOn Spoke: Create fix to ensure first letters of texter first & last name are always capitalized
Currently, if a texter creates a Spoke account like this "erik kirkstein" instead of like this "Erik Kirkstein", the texter's name does not appear in alphabetical order in the texter menu of a campaign, but instead at the bottom of the list.

it would be ideal if regardless of how a texter enters their name during the Spoke account setup process, that their name always appears in Spoke with the first letter of the first and last name capitalized.

### PR DESCRIPTION
**Pull request by azuzunaga**
Resolves MoveOnOrg/Spoke#948 and MoveOnOrg/Spoke#974.

This fix does not apply retroactively, only to campaigns created after the fix is deployed. I've also sorted the users in the "People" menu by first name. Let me know if both are OK.

An alternative would be to capitalize user's first and last name on account creation, and then create a migration to change the existing data in the DB. Let me know if this would be a better approach.

### CODE REVIEW

File: src/server/api/lib/utils.js
Comment (schuyler1d) [inline src/server/api/lib/utils.js]: this will fail on empty fields, so you'll need to guard against that.

File: src/server/api/organization.js
Comment (schuyler1d) [inline src/server/api/organization.js]: I think we should separate this from the captilize PR.  If we were going to do this, I think it would be better to change sorting in `buildUserOrganizationQuery` and make it `orderBy('first_name DESC')`  otherwise it could mess up pagination (if/when we have it)

**Review by schuyler1d — CHANGES_REQUESTED** [review]
I think I would rather just uppercase names on initial user save and then we don't have to muck in all these other places.  If we do, then it should just be `displayName` and then we can rely on that to show whatever we want correctly.

File: src/server/api/lib/utils.js
Comment (azuzunaga) [reply src/server/api/lib/utils.js]: ++

File: src/server/api/organization.js
Comment (azuzunaga) [reply src/server/api/organization.js]: ++

**Comment by azuzunaga** [comment]
> I think I would rather just uppercase names on initial user save and then we don't have to muck in all these other places. If we do, then it should just be `displayName` and then we can rely on that to show whatever we want correctly.

Sounds good. I'll update the user creation process and write a migration to change existing records.

**Comment by joemcl** [comment]
hey @azuzunaga how can we move this along? thanks.

**Comment by azuzunaga** [comment]
@joemcl, users' first and last names are already being capitalized on creation. This PR adds a migration to capitalize existing users' first and last names, and works with sqlite or postgres.

To move this along we need a PR review.

Note: Newest changes were force pushed because branch was very outdated and contained duplicated/reverted commits.

**Comment by azuzunaga** [comment]
@joemcl, I saw your comments on #974 and added a change so that users' first and last name are also capitalized when a user changes their name. Since we already do this on account creation I don't think it's too big of an issue.

**Comment by joemcl** [comment]
@azuzunaga thanks did not realize new users first and last now capitalized on creation....

**Comment by ibrand** [comment]
@azuzunaga and @schuyler1d what's the status of this PR? Were the requested changes made?

**Comment by azuzunaga** [comment]
hey @ibrand! I just fixed the latest merge conflict. This PR is ready for review.

**Comment by ibrand** [comment]
@schuyler1d Could you give this a final review? Looks like you had requested changes initially and I want to make sure we're all good to go. Thanks everyone!

File: src/migrations/index.js
Comment (schuyler1d) [inline src/migrations/index.js]: we'll need to remove this migration with #1154 (maybe just make it a separate PR, or don't actually change old campaigns, and skip it)

**Review by schuyler1d — DISMISSED** [review]
modulo removing the migration :+1:

File: src/migrations/index.js
Comment (azuzunaga) [reply src/migrations/index.js]: Thanks @schuyler1d, I'll remove the migration later today and push up the changes.

@ibrand I'll leave it up to you to decide whether we add the migration as a separate PR or if we just skip it. Thanks!

**Comment by azuzunaga** [comment]
Migration has been removed.

**Comment by ibrand** [comment]
@azuzunaga Awesome. I'll follow up once #1154 is merged since we're switching to new types of migrations so you can add a follow-up migration backfill PR. Thank you for your work on this!

**Comment by azuzunaga** [comment]
Thank you, @ibrand. And sounds good!

**Comment by shakalee14** [comment]
Merged with https://github.com/MoveOnOrg/Spoke/commit/2032a9b1004b41a4622896df9b0bb98ee7b136ed

**Pull request merged**
